### ISSUE TITLE
Define a `base-uri` Content Security Policy meta tag

### ISSUE THREAD

**Issue by maxime-rainville** [silverstripe/silverstripe-framework#10655]
We received a security report highlighting that a malicious user could potentially perform a _Host header cache poisoning_ to cache a malicious value in the  `<base>` tag. After analysis, we concluded that this behaviour would be a variation on a [Request Hostname Forgery](https://docs.silverstripe.org/en/4/developer_guides/security/secure_coding/#request-hostname-forgery) which we have historically considered to be server misconfiguration.

While discussing the issue with the reporter, it was suggested that we should consider adding "base-uri" Content-Security-Policy to further mitigate this attack vector.

Read [_CSP: base-uri_ on Mozilla Dev Docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/base-uri) for more details.

The TLDR is:
```html
<meta http-equiv="Content-Security-Policy" content="base-uri 'self'" />
<-- 
   Because of the above meta tag, browsers will not want to honour the 
   base tag if the page was not serve from "https://evil-hacker.net/"
-->
<base href="https://evil-hacker.net/" />
```

In theory, we could add that meta tag by default and it should work in the vast majority of cases. Otherwise, we could encourage devs to enable it for themselves as a security best practice.

**Comment by michalkleiner** [comment]
What would happen if you had defined CSP via this meta tag and e.g. via .htaccess or vhost config? Which takes precedence?

**Comment by GuySartorelli** [comment]
I think it definitely has to be configurable if we do include it by default. I also think it's a good thing to have, though would probably have to be disabled by default if not introduced in a major release.
In the meantime adding documentation about it to our security best practices doc would be a good step.

**Comment by GuySartorelli** [comment]
Labelling as enhancement for now until/unless we rule out adding some code implementation.
